client/landing/stepper/README.md

```diff
@@ -1,106 +1,387 @@
+
+**Note:** We have an example flow you can use as a reference [here](/client/landing/stepper/declarative-flow/flows/00-example-flow/example.ts).
+
+#### Code example
 
 ```ts
-import type { StepPath } from './internals/steps-repository';
-import type { Flow } from './internals/types';
+// We define our steps collection upfront because it is a prerequisite of shaping the flow.
+async function initialize( calypsoReduxStore: Store ) {
+	const includeDomainsStep = await isTheMoonFull();
+	const hasAnySites = userHasAnySites( calypsoReduxStore.getState() );
 
-export const exampleFlow: Flow = {
-	initialize() {
-		return [ STEPS.DOMAINS, STEPS.DESIGN ];
-	},
-	useStepNavigation( currentStep, navigate ) {
-		const goBack = () => {
-			if ( currentStep === 'domain' ) {
-				navigate( 'design' );
-			} else {
-				navigate( 'domain' );
+	if ( includeDomainsStep ) {
+		return [ STEPS.UNIFIED_DOMAINS, STEPS.UNIFIED_PLANS, STEPS.PROCESSING ] as const;
+	}
+
+	if ( hasAnySites ) {
+		return [ STEPS.PICK_SITE, STEPS.UNIFIED_DOMAINS, STEPS.UNIFIED_PLANS, STEPS.PROCESSING ] as const;
+	}
+
+	// We need `as const` to promise TS that these steps won't change later.
+	return [ STEPS.UNIFIED_PLANS, STEPS.PROCESSING ] as const;
+}
+
+export const exampleFlow: FlowV2< typeof initialize > = {
+	/**
+	 * The name of the flow is what appears in the pathname. It must be unique.
+	 * This flow will be under /setup/my-flow.
+	 */
+	name: 'my-flow',
+
+	/**
+	 * This flag must be `true` for signup flows (generally where a new site may be created), and should be `false` for other flows.
+	 * It controls whether we'll trigger a `calypso_signup_start` Tracks event when the flow starts.
+	 */
+	isSignupFlow: true,
+	initialize,
+	/**
+	 * This hook is the control unit of your flow. It is where:
+	 * 1. You handle and use the information submitted by the steps.
+	 * 2. You decide to navigate to which steps when.
+	 * 3. You manage the state of the flow.
+	 */
+	useStepNavigation( currentStepSlug, navigate ) {
+		// This hook should be enough to manage and persist all the state your flows needs.
+		const { get, set } = useFlowState();
+
+		// This function handles the submission event of every step in the flow.
+		const submit: SubmitHandler< typeof initialize > = async ( submittedStep ) => {
+			const { slug, providedDependencies } = submittedStep;
+			switch ( slug ) {
+				case 'domains': {
+					// Here we have the data submitted by the domains step.
+					// `providedDependencies` will be fully-typed and should contain the exact types of the domain step submission.
+					const domainItem = providedDependencies.domainItem;
```

Comment:
If I understand this correctly, the idea is that the flow will do something with this `domainItem`, e.g. save it into the Redux state, or update the site options, or whatever.

(Maybe I'm just not sure what "provided dependencies" mean. 😅 )

If that's the case, then I'd add an example of that here.
I can see that there's a `createSiteId` example in the `processing` step, but maybe it's too late. 🤔

Reply:
Great idea. Will do.